### DISCUSSION TITLE
Oqtane Module Development

### DISCUSSION THREAD

**Discussion by cherylbearden** [oqtane/oqtane.framework#948] [Q&A] [answered]
HI, I am going thru the Oqtane Module Development Series and am in Chapter 2 and trying to import text from the import.txt file. I'm getting an error message  saying that the content is not formatted correctly and am including a picture.  Thanks very much for the Module Development series!  It is very helpful. Note, in my installation, I named my project DNF.Project instead of DNF.Projects, so I changed the name in the text file. 
![image](https://user-images.githubusercontent.com/5729948/100227701-89ddad80-2eef-11eb-91e8-5f1decc4ec19.png)

**Comment by sbwalker** [comment, marked as answer]
@cherylbearden which version of Oqtane are you using - Oqtane 1.x on .NET Core 3.1 or Oqtane 2.x on .NET 5? And did you create your module using the Module Creator - or are you using the code in the https://github.com/oqtane/DNF.Projects repo?

**Reply by cherylbearden** [reply]
Hi Shaun, I am using Oqtane 2.x on .NET 5.  I originally built a new 'DNF.Project' project with the Module Creator, using the External template, following the Oqtane Module Development series. I also downloaded a copy of the 'DNF.Projects' code from https://github.com/oqtane/DNF.Projects, but did not install this module yet, as I was trying to build up the module by hand. When I got to Chapter 2, where data is pulled in from the file import.txt, I tried to import this data into my newly created 'DNF.Project' file, and I used the import.txt file that I found in the download from https://github.com/oqtane/DNF.Projects.  I only changed the name from 'DNF.Projects' to 'DNF.Project'.  Then I got the Error Importing Content.  So I found the dbo.DNF.Project table in my database, and noticed that it did not have all of the fields that were shown in the import.txt file.  So I tried adding in the missing fields and trying the import again, but I still got the same error.  Thanks very much.

**Reply by sbwalker** [reply]
Although the DNF.Projects project was initially created using the Module Creator, a number of enhancements were made to it so that it is capable of tracking Github project activity data. There is an additional database table which is where the import.txt data is imported. So you cannot use that import.txt file in a new module created using Module Creator. 

What you can do if you want to test import/export is create a new module using Module Creator, run the module and enter some data, export the data and modify it, and then import it.

**Reply by cherylbearden** [reply]
Ok, well, if I want to just pull in the existing DNF.Projects module into my Oqtane installation  ( I am using Oqtane 2.x on .NET 5 ), will that work, or or will the DNF.Projects project need to be upgraded first? Thanks again.

**Comment by cherylbearden** [comment]
I was able to upgrade the DNF.Projects module to .net5 by following this online example: https://www.youtube.com/watch?v=W1jP4q-7ALk and comparing the *.csproj files with *.csproj files of a sample project that I built using the Module Creator using Oqtane 2.x.  Once I did that, I added some xcopy statements to the  debug.cmd file so that Oqtane could find the .net5.0 and it seems to work.  Thanks again. :)

**Reply by cherylbearden** [reply]
Once the DNF.Project module was upgraded, and compiled to copy files into my Oqtane installation, I was able to go back to the module and import the data from the import.txt file with no errors.  So the data is showing up now in the [dbo].[DNFProjectActivity] table and shows up in the Project module.
